build.gradle

```diff
@@ -12,8 +12,8 @@ apply plugin: 'org.jetbrains.intellij'
 apply plugin: 'java'
 
 compileJava {
-    sourceCompatibility = 1.6
-    targetCompatibility = 1.6
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
```

Comment:
Could you tell why? Is it limitation from Android Studio?

I don't have much experience in Android Studio plugins development, but I managed to build your plugin with Java 1.7 (with some warnings) and run it successfully on Android Studio. Even creating a new IntelliJ Platform Plugin project on Java 1.8 worked fine for me.